src/realm/object-store/shared_realm.hpp

```diff
@@ -378,6 +378,25 @@ class Realm : public std::enable_shared_from_this<Realm> {
         return !m_transaction && !m_coordinator;
     }
 
+    /**
+     * Deletes the following files for this Realm's configuration if they exist:
+     * - the Realm file itself
+     * - the .log file and its legacy versions: .log_a and .log_b
+     * - the .note file
+     * - the .management folder
+     *
+     * The .lock file for this Realm cannot and will not be deleted as this is unsafe.
+     * If a different process / thread is accessing the Realm at the same time a corrupt state
+     * could be the result and checking for a single process state is not possible here.
+     *
+     * @return Yes, if all files could either be deleted or did not exist.
+     *         No if at least one existing file coule not be deleted or the Realm is still open.
+     *
+     * @throws PermissionDenied if the operation was not permitted.
+     * @throws AccessError for any other error while trying to delete the file or folder.
+     */
+    static bool delete_files(const Config& config);
```

Comment:
@ironage Nikola pointed out that this needs to be static so that we don't need to open a Realm just so we can delete it's files.